### DISCUSSION TITLE
Show current streak

### DISCUSSION THREAD

**Discussion by bbilly-git** [iSoron/uhabits#690] [Feature Requests]
I haven't been able to find a way to show the current streak for my habits.
The Best streaks stat screen only show the past best streaks.

Would love to have a way to feel current progress by displaying the current streak in number of days. That's a feature that most apps (even not habit trackers) do have to entice users to repeat their behaviour, see that number grow and not break the chain. The habit score is great but doesn't convey that feeling of building a chain/streak.
Ideally I'd like to easily see it across all my habits in one screen (it could be in a widget, or on the main screen alongside each of my habits)

**Comment by iSoron** [comment]
Thank you for the feature request, @bbilly-git. In the past, we used to show the current streak at the top of the "Best Streaks" chart, but this was removed in newer versions of the app. If anyone else is interested in this feature, please leave your comments here or give this thread a thumbs-up, so we know there's user demand for it. Also, please feel free to leave suggestions here on how could we show the current streak in the user interface.

**Comment by viniciussbs** [comment]
This is the most important information about the habit for me, because I don't want to break the chain. The second one is the closest "best streak" to my current streak, because, when I break the chain, I see my best streaks as milestones.

**Comment by jean3x7** [comment]
Please bring back current streak!

**Comment by SG60** [comment]
It would be great if this could be added. It's probably the most important info for encouraging me to complete a habit!

**Comment by baincd** [comment]
I would be interested in this as well.

In fact, I've actually [implemented this in my own fork](https://github.com/baincd/android-habits/commit/8b89243eedad8b3d5e721e1347c7e0f6d981ceb2) because I wanted this feature.  It's not an optimal implementation, just a way to get it working.  What is does is:

- The current streak is always listed first in the Best streaks
  - I defined the current streak as the streak that includes yesterday, so I can see how long of a streak I would be breaking if I don't continue the habit today.
- If there is no current streak, then show a streak of zero

**Reply by SG60** [reply]
Do you think this could be merged into this repo? Behind an option perhaps?

Although not a true solution, it's definitely better than nothing.

**Reply by baincd** [reply]
I don't mind working on a releasable implementation of this when I have some time to work on it, but only if the maintainers are interested in having this added.  I would recommend this being an option, as not everyone may want this feature.

**Comment by ZDUAA08** [comment]
I would love to see a feature where I can track a streak for all of my habits not just individual habits

**Comment by ke352802081770314** [comment]
maybe I am confused about what we are talking about - assuming it is when I tap on a (non-numerical) habit and scroll down below calendar and above the frequency different-sized dots diagram?
 
@iSoron I see the streak ending today or yesterday (depending on whether I already added the current day, which for some habits is only due in the evening) as the newest entry in "Beste Serien" (only exists in non-numerical habits) and I'm on version 2.0.2 - to be honest I should say "best streaks" is a misnomer because they are not sorted by duration but just the most recent streaks as far as I can tell, but that seems more relevant and helpful to me so that I can see the development. so the best option to me would be to rename the heading to "recent streaks" or "latest streaks" and leave it exactly as it is (or was).

**Reply by baincd** [reply]
As of 2.0.3, The Yes/No Streaks only show the 10 longest streaks (displayed in chronological order).

Here is an example with one streak of each length 3 through 15, with the streak of 3 being the current streak.  Under Best Streaks, you will current streak of 3 is not displayed - only the streaks of length 5-14 are displayed:
![20220320_224336](https://user-images.githubusercontent.com/5075659/159199007-ed72b4d2-befa-450a-a5ec-2d4be17a359e.jpg)

**Comment by Noracsa2000** [comment]
I just wanted to request the same thing, then found this discussion.

Showing the growing current streak gives a much better motivation when using the streaks widget.

The current implementation just shows a static history of past streaks without ever changing, until, eventually, the current streak appears within the best streaks.

Always showing the current streak on top would me much appreciated.

**Comment by ke352802081770314** [comment]
I will simply not update from my currently installed version (2.0.2) until I see this was restored.

**Comment by oliveralonzo** [comment]
This is super important to me, too! Streaks are super motivating. My suggestion would be to add the option to show either the circle visualizing the current score (as it currently is) or the current streak on the left of each habit in the main page. The current streak is much more digestable and motivating to me than the score, so I’d be okay with replacing that, and it looks like there’s enough space to show a number. But I’d of course make it an option since some people might like the score. 

A rough screenshot of what I’m thinking below (circle for consistency, but not needed).  

![Screenshot_20230330-111723](https://user-images.githubusercontent.com/19820425/228902452-9744510c-51ed-4d24-b793-e5e3f7d49aae.png)

**Reply by obbardc** [reply]
This is exactly how I think it should be shown!

**Reply by GeorgeD88** [reply]
I absolutely agree!

**Reply by ZornHadNoChoice** [reply]
You can show both! You already have a circle in your screenshot, much like the current score circle.

**Comment by viniciussbs** [comment]
Something that would be realy nice is having the current streak in this Best Streaks section but with the option to sort by score. So you can see the current streak and your own streaks will be like new challenges.

Or maybe a new section. In the example, the current streak is 8 and next milestones are: 9, 12 and 14. So you are challenging yourself. This way you can have small victories until you make a new best streak.

![next-streaks](https://user-images.githubusercontent.com/245791/229200453-dddb2791-cbaa-4085-bda9-ddcef8bb29b3.png)

**Comment by cosminp7** [comment]
This is also very important to me too to the point where it is a deal breaker. I would like to see that for both Yes/No and numerical habits.

**Comment by nickfrescas** [comment]
If not in the best streaks section, it could also be included in the overview section or just as it's own line below the overview somewhere.

**Comment by ZornHadNoChoice** [comment]
The PR adding best streaks to measurable habits has finally been merged. Can we get this next, please?
https://github.com/iSoron/uhabits/discussions/690#discussioncomment-2121422

**Reply by ZornHadNoChoice** [reply]
Also, adding the current streak to the top of the best streaks is better imo, since it allows the streaks widget to display both the current and best streaks, instead of having two widgets taking up space.

**Comment by testarossa47** [comment]
I am totally missing this option.
Actually it would be great to have a widget with the current streak to keep you motivated not to lose it.

If you could decide between percentage and current streak. Or  show both since I find both useful but for some specific habits I find the streak to be better.

**Comment by TeoSchweitzer** [comment]
Missing this feature too, especially in the form of a widget with a big streak number on it. I'm moving on from Duolingo, it had that, I'd like to keep it.

**Reply by chintanKalkura** [reply]
I want the exact same thing!

**Comment by chintanKalkura** [comment]
Seeing your current streak is more motivating than the score. It creates the urge to keep the streak. I would like a widget just like the checkmark widget, which shows the current streak and we can mark done from the widget. It would also be good to get the current streak in the reminder.

I have made a widget that shows the current streak and the widget can be use mark done. The changes are in my fork. 
https://github.com/chintanKalkura/uhabits/tree/feature/current-streak-widget

<img width="1002" height="1177" alt="Screenshot_20260226_183408" src="https://github.com/user-attachments/assets/b7012cbd-0174-43d0-8e0f-43d2e361a60c" />

**Comment by srix55** [comment]
I was searching through the settings to find a disabled current-streak option. The best streaks don't motivate me. In fact, it does the opposite. It's demotivating to see how high I need to climb to break a previous big streak and I don't even start.

**Comment by livwire** [comment]
I would also love this feature! Seeing my current streak is super motivating. Thanks for considering and fingers crossed it gets added. I think it would b perfect right above best streaks, in the same look as best streaks :)
